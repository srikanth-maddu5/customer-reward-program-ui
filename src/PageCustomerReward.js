import React from 'react';
import PropTypes from 'prop-types';
import { Layout, Result, Spin } from 'antd';

import TableCustomer from './TableCustomer';

const { Header, Content } = Layout;

function PageCustomerReward(props) {

  const { customerData, transactionData, spinner, error } = props;

  return (
    <Layout className='layout'>
      <Header style={{ backgroundColor: '#004165', textAlign: 'center' }}>
        <h2 style={{ color: 'whitesmoke' }}>Customer Reward Portal</h2></Header>
      <Layout>
        <Content className='site-layout-content'>
          <Spin spinning={spinner}>
          { error ? (
            <Result 
              status={'error'} 
              title={'Customer Data Retrival Failed!'}
              subTitle={error}
            />
          ) : (
            <TableCustomer 
              customerData={customerData} 
              transactionData={transactionData} 
            />
          )}
          </Spin>
        </Content>
      </Layout>
    </Layout>
  );
}

PageCustomerReward.prototype = {
  customerData: PropTypes.array,
  transactionData: PropTypes.array,
  spinner: PropTypes.bool,
  error: PropTypes.string
}

export default PageCustomerReward;
