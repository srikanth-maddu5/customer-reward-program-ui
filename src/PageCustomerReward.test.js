import { render, screen } from '@testing-library/react';
import PageCustomerReward from './PageCustomerReward';

test('test page customer table component error message', () => {
    const { asFragment } = render(<PageCustomerReward 
        customerData={[]}
        transactionData={[]}
        spinner={false}
        error={'unable to retrive data'}
    />);

    // check page header
    expect(screen.getByRole('heading')).toHaveTextContent('Customer Reward Portal');

    // check error message 
    expect(asFragment()).toMatchSnapshot(`<div class="ant-result-subtitle">unable to retrive data</div>`);
});