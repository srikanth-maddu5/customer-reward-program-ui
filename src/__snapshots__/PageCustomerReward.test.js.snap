// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test page customer table component error message: <div class="ant-result-subtitle">unable to retrive data</div> 1`] = `
<DocumentFragment>
  <section
    class="ant-layout layout"
  >
    <header
      class="ant-layout-header"
      style="background-color: rgb(0, 65, 101); text-align: center;"
    >
      <h2
        style="color: whitesmoke;"
      >
        Customer Reward Portal
      </h2>
    </header>
    <section
      class="ant-layout"
    >
      <main
        class="ant-layout-content site-layout-content"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-result ant-result-error"
            >
              <div
                class="ant-result-icon"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="ant-result-title"
              >
                Customer Data Retrival Failed!
              </div>
              <div
                class="ant-result-subtitle"
              >
                unable to retrive data
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</DocumentFragment>
`;
